### PR TITLE
Added Wiki Pages, Documentation, and License to ReadME.md

### PR DESCRIPTION
**Pull request by pl9870**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+About the Allergy API Module
+===
+The Allergy API is a Java-based API that handles patient data in OpenMRS related to allergies. More specifically, the main motivations
+behind this API are to easily add allergy objects to patients, to change certain aspects of allergy objects for a patient such as severity and allergens in an organized fashion
+(through variables such as allergy ID), and to compare allergies with each other to see if they have similar properties or characteristics.
+
+Developer Documentation
+---
+
+Allergy.java
```
Comment (suthagar23) [inline README.md]: We don't need this much of the developer information for each JAVA file. As we need to change this ReadMe when we change those functions. We have method level comments to show the necessary docs. 

Can you simply mention some common facts for Developer documentation? Like JIRA links and other folder structure.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+---
+| Page | Description |
+| ---- | ----------- |
+| [Allergies Reference](https://wiki.openmrs.org/pages/viewpage.action?pageId=48857177) | Refers to much of the functionality of the Allergy API|
+| [Synchronizing Allergies](https://wiki.openmrs.org/display/projects/Allergy+synchronization) | Describes OpenMRS Fields and other fields|
+| [Adding Rest Resources to Core](https://issues.openmrs.org/browse/RA-1036) | Description of how to add REST resources to OpenMRS core. |
+| [2.5 Issue Tracking](https://wiki.openmrs.org/display/RES/Reference+Application+2.5+Release+Issue+Tracking) | Description of issues for various modules in OpenMRS |
+
+
+Contribute
```
Comment (suthagar23) [inline README.md]: This should be heading

File: README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+---
+OpenMRS is a highly active and engaged open-source community who would love to have more help!
+To contribute to APIs like this and more, sign up for an OpenMRS ID in the following link: 
+
+https://openmrs.org/
+
+
+License
```
Comment (suthagar23) [inline README.md]: This should be heading

File: README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+## Developer Documentation
+
+### Getting Started
+* The actual API is meant to be used with the allergyUI module, as this API is meant to act like a framework. This API is simply a module of OpenMRS's main application, and thus is not meant to be used in isolation aside from testing purposes (like bugs, etc.). More about the allergyUI module can be found [here](https://github.com/openmrs/openmrs-module-allergyui)
+* API References can be found in the actual src files in the repository as javadoc comments specifically what each method does.
+* On the Allergies Reference wiki page and OpenMRS Talk, many developers have said that this API has been implemented
+in the OpenMRS platform under patient as a subresource.
+
+### JIRA Issues
```
Comment (suthagar23) [inline README.md]: Remove this, as we need to change it each day

**Comment by suthagar23** [comment]
@sherrif10  Yes, as it needs more work, We can create another ticket for that

File: README.md (hunk 2 above)
Comment (sherrif10) [reply README.md]: i think you can abandon installation of allergy module for now, concentrate on allergyapi as you did, good work

**Comment by pl9870** [comment]
thank you, will do

File: README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+:maple_leaf: Maven and Java 8 need to be installed to successfully build and run 
+the tests. :maple_leaf:
+
+## Developer Documentation
+
+### Getting Started
+* The actual API is meant to be used with the allergyUI module, as this API is meant to act like a framework. This API is simply a module of OpenMRS's main application, and thus is not meant to be used in isolation aside from testing purposes (like bugs, etc.). More about the allergyUI module can be found [here](https://github.com/openmrs/openmrs-module-allergyui)
+* API References can be found in the actual src files in the repository as javadoc comments specifically what each method does.
+
+## JIRA Issues
```
Comment (suthagar23) [inline README.md]: As I mentioned before, you don't need to include this.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+
+<img src="https://talk.openmrs.org/uploads/default/original/2X/f/f1ec579b0398cb04c80a54c56da219b2440fe249.jpg" alt="OpenMRS"/>
+
+About the Allergy API Module
```
Comment (suthagar23) [inline README.md]: OpenMRS Allergy API Module

**Comment by suthagar23** [comment]
Seems good. 
Please follow our pull request tips to merge this PR : https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Pull request closed without merging**